routers/api/v1/repo/issue_reaction.go

```diff
@@ -0,0 +1,394 @@
+		return
+	}
+	_, err = reactions.LoadUsers()
+	if err != nil {
+		ctx.Error(500, "ReactionList.LoadUsers()", err)
+		return
+	}
+
+	var result []api.ReactionResponse
+	for _, r := range reactions {
+		result = append(result, api.ReactionResponse{
+			User:     r.User.APIFormat(),
+			Reaction: r.Type,
+			Created:  r.CreatedUnix.AsTime(),
+		})
+	}
+
+	ctx.JSON(200, result)
+}
+
+// PostIssueCommentReaction add a reaction to a comment of a issue
+func PostIssueCommentReaction(ctx *context.APIContext, form api.EditReactionOption) {
+	// swagger:operation POST /repos/{owner}/{repo}/issues/comments/{id}/reactions issue issuePostCommentReaction
+	// ---
+	// summary: Add a reaction to a comment of a issue comment
+	// consumes:
+	// - application/json
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: id
+	//   in: path
+	//   description: id of the comment to edit
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// - name: content
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/EditReactionOption"
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/ReactionResponse"
+	changeIssueCommentReaction(ctx, form, true)
+}
+
+// DeleteIssueCommentReaction list reactions of a issue comment
+func DeleteIssueCommentReaction(ctx *context.APIContext, form api.EditReactionOption) {
+	// swagger:operation DELETE /repos/{owner}/{repo}/issues/comments/{id}/reactions issue issueDeleteCommentReaction
+	// ---
+	// summary: Remove a reaction from a comment of a issue comment
+	// consumes:
+	// - application/json
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: id
+	//   in: path
+	//   description: id of the comment to edit
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// - name: content
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/EditReactionOption"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/empty"
+	changeIssueCommentReaction(ctx, form, false)
+}
+
+func changeIssueCommentReaction(ctx *context.APIContext, form api.EditReactionOption, isCreateType bool) {
+	comment, err := models.GetCommentByID(ctx.ParamsInt64(":id"))
+	if err != nil {
+		if models.IsErrCommentNotExist(err) {
+			ctx.NotFound(err)
+		} else {
+			ctx.Error(500, "GetCommentByID", err)
+		}
+		return
+	}
+
+	err = comment.LoadIssue()
+	if err != nil {
+		ctx.Error(500, "comment.LoadIssue() failed", err)
+	}
+
+	if comment.Issue.IsLocked && !ctx.Repo.CanWrite(models.UnitTypeIssues) && !ctx.User.IsAdmin {
```

Comment:
Shall be OR instead of AND? Currently you can post reaction if issue is not locked even if don't have write permission.

Reply:
Good  👀 I should fix this

EDIT: This should work:
- A := Admin
- L := Loked
- W := Write Permission
Allowed to Reacht = A || !L && W
!R = !A && !(!L && W)
!R = !A && L && !W

Wich is the current Solution ... so it should work @davidsvantesson

Reply:
@davidsvantesson its already in usage: https://github.com/go-gitea/gitea/blob/d67220680015068e3fbda7d94345630102469ca5/routers/api/v1/repo/issue_comment.go#L187

Reply:
I have read up on the locking feature, so writers can still make comments when issue locked so then this code makes sense.